src/main/java/umk/mat/pajda/ProjektZespolowy/DTO/TipDTO.java

```diff
@@ -0,0 +1,35 @@
+package umk.mat.pajda.ProjektZespolowy.DTO;
+
+import jakarta.validation.constraints.FutureOrPresent;
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Positive;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+import java.time.LocalDateTime;
+
+@Setter
+@Getter
+@NoArgsConstructor
+@AllArgsConstructor
+public class TipDTO {
+
+    @NotNull
+    @Positive
+    private Integer id;
+
+    private String currency;
+
+    @NotNull
+    private Float amount;
+
+    @FutureOrPresent
+    private LocalDateTime paymentTime;
+
+    private String paidWith;
+
+    @NotNull
+    private UserDTO user;
```

Comment:
I think it should have two entities linked here. The anonymous payer (a customer) and waiter, who needs to get paid, right?

Reply:
Entity user means waiter. We don't save client in database, so I didn't write their there